react/Payment.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useRef, useEffect, useCallback } from 'react'
+import { useSSR } from 'vtex.render-runtime'
+
+if (window && window.document) {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  var postRobot = require('post-robot')
+}
+
+interface Card {
+  cardNumber: string
+  cardName: string
+  cardDate: string
+  cardCvv: string
+}
+
+const IFRAME_APP_VERSION = '0.1.1'
+const LOCAL_IFRAME_DEVELOPMENT = false
+
+let iframeURL = `https://io.vtexpayments.com.br/card-form-ui/${IFRAME_APP_VERSION}/index.html`
+
+if (LOCAL_IFRAME_DEVELOPMENT) {
+  const PORT = 3000
+  iframeURL = `https://checkoutio.vtexlocal.com.br:${PORT}/`
+}
+
+const Payment: React.FC = () => {
+  const [cardData, setCardData] = useState<Card | null>(null)
+  const iframeRef = useRef<HTMLIFrameElement>(null)
+
+  const isSSR = useSSR()
+
+  const setupIframe = useCallback(() => {
+    const stylesheetsUrls = Array.from(
+      document.head.querySelectorAll<HTMLLinkElement>('link[rel=stylesheet]')
+    ).map(link => link.href)
+
+    postRobot.send(iframeRef.current!.contentWindow, 'setup', {
+      stylesheetsUrls,
+    })
+  }, [])
+
+  useEffect(() => {
+    const listener = postRobot.on('card', ({ data }: { data: Card }) => {
+      setCardData(data)
+      return {
+        status: 'ok',
+      }
+    })
+    return () => listener.cancel()
+  })
```

Comment:
Are you sure you want to register a listener at every update? (since this `useEffect` has no second argument) Also, this side-effect is not SSR-safe

Reply:
what do you mean with "SSR-safe"?

Reply:
I forget this detail

Reply:
@lucasecdb I mean that because of lines 4-6 on this same file, `postRobot` is only defined when this component is rendered client-side. I just fear that by executing this `useEffect` as-is, we might have a TypeError at SSR. My doubt is, by doing
```
if (isSSR) {
  return null
}
```
alone, do we prevent the side-effects from executing when this component is mounted?

Reply:
@arturpimentel `useEffect`, and all lifecycle methods, are not executed during SSR.

Reply:
It's all right, then 😁